File: src/main/java/fr/univ_smb/isc/m1/brutocollinus/infrastructure/persistence/entity/Stuff.java
package fr.univ_smb.isc.m1.brutocollinus.infrastructure.persistence.entity;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.OneToOne;

@Entity
public class Stuff extends Nameable {
    @OneToOne(cascade = CascadeType.ALL)
    private FightStatistics fightStatistics;

    public Stuff() {
        // JPA
    }

    public Stuff(String name, FightStatistics fightStatistics) {
        super(name);
        this.fightStatistics = fightStatistics;
    }

    public FightStatistics fightStatistics() {
        return this.fightStatistics;
    }
}

File: frontend/src/components/Stuff.js
import React from 'react'
import FightStatistics from './FightStatistics.js';

export default function Stuff(props) {
    const stuff = props.stuff;

    return (
        <div>
            {stuff.name}
            <FightStatistics fightStatistics={stuff.fightStatistics}/>
        </div>
    );
}